src/parser/parser-models.ts

```diff
@@ -39,3 +39,47 @@ export interface DomHtmlNode {
 export interface ParseResult {
   children: DomNode[];
 }
+
+type DeliverObjectElementAttributes = {
+  'data-rel': 'component' | 'link';
+  'data-type': 'item';
+  'data-codename': string;
+  'data-id': undefined;
```

Comment:
we need to explicitly say its undefined? :D

Reply:
either that or preferably `never`, which it was originally but I modified it during refactoring for some reason... I changed it back. it's to clarify that each property is only present in its respective representation (DAPI vs MAPI) in the subsequent union:

```
export type ObjectElementAttributes = 
  | DeliverObjectElementAttributes 
  | ManagementObjectElementAttributes;
```